### PR TITLE
Add #97640: Added options to enable codelens for diff editors

### LINKED ISSUE #97640: Support CodeLens in diff editor
Related to issue [monaco-editor #1909](https://github.com/microsoft/monaco-editor/issues/1909), it would be great to be able to add code lenses to the diff editor (original and/or modified) but they are hard coded to be disabled in diff editors (see [here](https://github.com/microsoft/vscode/blob/6d70b4f04271dfd5c9f0693a4c7952fdc9c72b7d/src/vs/editor/browser/widget/diffEditorWidget.ts#L1051)).

It should be possible to enable CodeLens in the original/modified editor with editor options.

### PR DESCRIPTION
**Pull request by ludokx**
This PR fixes #97640.

### Description
This PR adds new options `originalCodeLens` and `modifiedCodeLens` to enable CodeLens in the original/modified diff editors (hard coded to `false` before this PR).

### How to Test
As described in the monaco-editor [documentation](https://github.com/microsoft/monaco-editor/blob/master/CONTRIBUTING.md), `vscode` and `monaco-`editor must be siblings.

Once `vscode` is [built](https://github.com/Microsoft/vscode/wiki/How-to-Contribute#build-and-run) with `yarn watchd`, go into `monaco-editor`, run `npm run build-website` and then `npm run simpleserver`.

Pick an HTML file under `test/playground.generated/` and change to the source below, then open in the browser. I modified `extending-language-services-codelens-provider-example.html`.
```
var originalModel = monaco.editor.createModel("just some text\n\nHello World\n\nSome more text", "plaintext");
var modifiedModel = monaco.editor.createModel("just some Text\n\nHello World\n\nSome more changes", "plaintext");

var editor = monaco.editor.createDiffEditor(document.getElementById("container"), { modifiedCodeLens: true });
editor.setModel({
	original: originalModel,
    modified: modifiedModel,
});

var commandId = editor.addCommand(0, function () {
    alert('code lens is working!');
}, '');

monaco.languages.registerCodeLensProvider('plaintext', {
    provideCodeLenses: function (model, token) {
        return {
            lenses: [
                {
                    range: {
                        startLineNumber: 1,
                        startColumn: 1,
                        endLineNumber: 2,
                        endColumn: 1
                    },
                    id: "First Line",
                    command: {
                        id: commandId,
                        title: "First Line"
                    }
                }
            ]
        };
    },
    resolveCodeLens: function (model, codeLens, token) {
        return codeLens;
    }
});
```

Open your browser at:
http://127.0.0.1:8080/monaco-editor/test/playground.generated/extending-language-services-codelens-provider-example.html?editor=src

For `editor` on the right-hand side, select `src` to run the code you build from vscode.

The code lens shows up in the first line of the modified editor.

### CODE REVIEW

**Comment by alexdima** [comment]
@ludokx Please sign the Contributor License Agreement, so I can take a look at this -- see the above comment

**Comment by ludokx** [comment]
@alexdima done, sorry needed to clarify some things first. What about the VS Code pipeline failure though, seems to be something intermittent and unrelated. How can it be re-run?

**Comment by ludokx** [comment]
@alexdima quick follow to see if there's anything I can do to get the VS Code check to pass?

File: src/vs/monaco.d.ts [hunk 1]
```diff
@@ -3172,6 +3172,16 @@ declare namespace monaco.editor {
 		 * Defaults to false.
 		 */
 		originalEditable?: boolean;
+		/**
+		 * Original editor should be have code lens enabled?
+		 * Defaults to false.
+		 */
+		originalCodeLens?: boolean;
+		/**
+		 * Original editor should be have code lens enabled?
```
Comment (MehaKaushik) [inline src/vs/monaco.d.ts:3181]: typo: should be "Modified editor" here

File: src/vs/editor/common/config/editorOptions.ts [hunk 2]
```diff
@@ -643,6 +643,16 @@ export interface IDiffEditorOptions extends IEditorOptions {
 	 * Defaults to false.
 	 */
 	originalEditable?: boolean;
+	/**
+	 * Original editor should be have code lens enabled?
+	 * Defaults to false.
+	 */
+	originalCodeLens?: boolean;
+	/**
+	 * Original editor should be have code lens enabled?
```
Comment (MehaKaushik) [inline src/vs/editor/common/config/editorOptions.ts:652]: same as below

File: src/vs/monaco.d.ts (hunk 1 above)
Comment (ludokx) [reply src/vs/monaco.d.ts]: oops, copy/paste... :/ good catch, thanks!

**Comment by ludokx** [comment]
@MehaKaushik thank for catching those typos. The VS Code pipeline failed again but I don't see how this could be related to my change, the default values should have no impact on prior behavior at all. Is there a way to kick off the pipeline again without creating a new commit?

**Comment by MehaKaushik** [comment]
> @MehaKaushik thank for catching those typos. The VS Code pipeline failed again but I don't see how this could be related to my change, the default values should have no impact on prior behavior at all. Is there a way to kick off the pipeline again without creating a new commit?

I believe anyone with write permissions on this repo would be able to run the pipelines for you.

File: src/vs/monaco.d.ts (hunk 1 above)
Comment (MehaKaushik) [reply src/vs/monaco.d.ts]: no problem

**Comment by ludokx** [comment]
@alexdima @MehaKaushik anything else I should do for this PR? Thanks

**Comment by alexdima** [comment]
Thank you!

**Pull request merged**